File: articles/automated-reasoning-symbolic-ai-and-hybrid-systems/java/SymbolicRuleCheck.java
import java.util.Set;

public class SymbolicRuleCheck {
    public static void main(String[] args) {
        Set<String> facts = Set.of("has_documentation", "logs_decisions");
        boolean fires = facts.contains("has_documentation") && facts.contains("logs_decisions");
        System.out.println("rule_fires=" + fires);
    }
}

File: articles/automated-reasoning-symbolic-ai-and-hybrid-systems/rust/symbolic_rule_check.rs
use std::collections::HashSet;

fn main() {
    let facts: HashSet<&str> = ["has_documentation", "logs_decisions"].iter().copied().collect();
    let premises = ["has_documentation", "logs_decisions"];
    let fires = premises.iter().all(|p| facts.contains(p));
    println!("rule_fires={}", fires);
}
